### ISSUE TITLE
Fresh install, can't run any react-native commands / Unexpected strict mode reserved word

### ISSUE THREAD

**Issue by istrasoft** [facebook/react-native#10976]
### Description

Followed this tutorial to install everything to use RN on Windows 8.1
https://facebook.github.io/react-native/docs/getting-started.html

### Reproduction

C:\WINDOWS\system32>react-native init helloworld

C:\Users\myuser\AppData\Roaming\npm\node_modules\react-native-cli\index.js:81
  let yarnVersion;
  ^^^
SyntaxError: Unexpected strict mode reserved word
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3

Any other react-native command leads to the same result

### Additional Information

* Node version [7.1.0]
* React Native version: [0.37.0]
* Platform: [Android]
* Operating System: [Windows 8.1]

**Comment by grabbou** [comment]
What's your `node` version?

**Comment by chadlinden** [comment]
I'm having the same issue. After running `react-native init tralala` I ran `react-native run-android` and `react-native run-ios` with `node` 7.0.0, 6.9.1 and finally 5.12.0 All three failed outputting `Uhandled JS Exception: TypeError`

**Comment by istrasoft** [comment]
node 0.10.24

**Comment by fannedalk** [comment]
Just had the same problem on macOS. I updated node to v7.0.0 and now it works

**Comment by tomazahlin** [comment]
Omg, problems, problems everywhere.... Even installing fresh react-native doesn't work. Seriously, we are in 2016, wake up Facebook!

**Comment by littledot** [comment]
I'm seeing the same issue on OSX 10.11.6
```
$ react-native init hello-rn

/.../.nvm/v0.10.32/lib/node_modules/react-native-cli/index.js:81
  let yarnVersion;
  ^^^
SyntaxError: Unexpected strict mode reserved word
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:906:3
```

What can I do to proceed?

**Comment by istrasoft** [comment]
funny how since i opened the issue many other people seem to have it :)

**Comment by fannedalk** [comment]
To clarify a bit how I got it working:
Updated node from version v0.12.7 with following commands:

sudo npm cache clean -f
sudo npm install -g n
sudo n stable

Im now on node v7.0.0.

Then I got some another problem: npm WARN locking Error: EACCES: permission denied 
on some lock file.

I then ran: npm cache clean

Now I was able to run react-native init AwesomeProject without problems

**Comment by istrasoft** [comment]
Problem is, on windows npm install -g n leads to a message saying N does not support the platform :) 

but thanks @fannedalk

**Comment by istrasoft** [comment]
Tried to uninstall node 7.1.0 and install node 6.8.0, same issue

**Comment by chadlinden** [comment]
So React uses a version of babel which prototypes `babel.typeof`, if your `.babelrc` doesn't have the proper preset, you'll have this problem. The proper preset, `"presets": ["react-native-stage-0"]` will load the correct babel preset. Then be sure to install that preset in your `package.json` 
```
  "dependencies": {
    "babel-preset-react-native-stage-0": "^1.0.1",
```

**Comment by istrasoft** [comment]
hello @chadlinden, thanks for the clear cache command, how does one install the proper preset in the package.json etc ?

**Comment by istrasoft** [comment]
Thanks @chadlinden that helped

**Comment by stevechui** [comment]
@fannedalk thanks you saved my day

**Comment by mkonicek** [comment]
Whoa looks like I should have used `var` instead of `let` in this file?
https://github.com/facebook/react-native/blob/master/react-native-cli/index.js

Tested on Mac OS with Node 6 and 7.

**Comment by mkonicek** [comment]
@chadlinden RN apps created by 'react-native init' have the following in package.json. Is that not sufficient?

    "devDependencies": {
      "babel-preset-react-native": "1.9.0",

**Comment by mogazheng** [comment]
same issue,
cause by multi version of nodejs in my windows PC,
edit the envionment variables,
use the new one and it works

**Comment by mkonicek** [comment]
https://github.com/facebook/react-native/pull/11166

**Comment by mkonicek** [comment]
Published an update (1.3.0): https://www.npmjs.com/package/react-native-cli

Please run 'npm install -g react-native-cli' to get it.
